### PR TITLE
Add GUI component icons and ID field. 

### PR DESCRIPTION
**Pull request by ZhangAnli**

### CODE REVIEW

File: src/main/resources/view/PawbookTheme.css [hunk 1]
```diff
@@ -308,7 +320,7 @@
     -fx-padding: 8 1 8 1;
 }
 
-#cardPane {
+#cardPaneDog {
```
Comment (kouyk) [inline src/main/resources/view/PawbookTheme.css:323]: is this a mistake? I don't think I see any cardPaneDog around

File: src/main/java/dog/pawbook/ui/MainWindow.java [hunk 2]
```diff
@@ -21,7 +21,7 @@
  * The Main Window. Provides the basic application layout containing
  * a menu bar and space where other JavaFX elements can be placed.
  */
-public class MainWindow extends UiPart<Stage> {
+public class MainWindow<FooTabController> extends UiPart<Stage> {
```
Comment (kouyk) [inline src/main/java/dog/pawbook/ui/MainWindow.java:24]: What does the generic parameter do?

File: src/main/java/dog/pawbook/ui/EntityCard.java [hunk 3]
```diff
@@ -38,13 +45,31 @@
...
+        this.entityImage.setFitHeight(50);
+        this.entityImage.setFitWidth(50);
+
+        if (entity instanceof Dog) {
+            this.entityImage.setImage(dogImage);
+        } else if (entity instanceof Owner) {
+            this.entityImage.setImage(ownerImage);
+        } else {
+            this.entityImage.setImage(programImage);
+        }
```
Comment (kouyk) [inline src/main/java/dog/pawbook/ui/EntityCard.java:71]: is it better if program is explicitly matched as well? then you get to do some defensive coding by asserting in the else

File: src/main/java/dog/pawbook/ui/EntityCard.java [hunk 4]
```diff
@@ -38,13 +45,31 @@
...
 
     /**
      * Creates a {@code EntityCard} with the given {@code Entity} and index to display.
      */
     public EntityCard(Entity entity, int displayedId) {
         super(FXML);
         this.entity = entity;
+
+        this.entityImage.setFitHeight(50);
+        this.entityImage.setFitWidth(50);
```
Comment (kouyk) [inline src/main/java/dog/pawbook/ui/EntityCard.java:63]: is there a way to specify this in fxml? I think it is generally better if parameters that don't need to be dynamically set be left out of the code

**Review by kouyk — COMMENTED** [review]
Generally I don't see any problems so long as the GUI works.

File: src/main/resources/view/PawbookTheme.css (hunk 1 above)
Comment (ZhangAnli) [reply src/main/resources/view/PawbookTheme.css:323]: Fixed.

File: src/main/java/dog/pawbook/ui/MainWindow.java (hunk 2 above)
Comment (ZhangAnli) [reply src/main/java/dog/pawbook/ui/MainWindow.java:24]: Was a work in progress, removed for now.

File: src/main/java/dog/pawbook/ui/EntityCard.java (hunk 3 above)
Comment (ZhangAnli) [reply src/main/java/dog/pawbook/ui/EntityCard.java:71]: Done.

File: src/main/java/dog/pawbook/ui/EntityCard.java (hunk 4 above)
Comment (ZhangAnli) [reply src/main/java/dog/pawbook/ui/EntityCard.java:63]: As of now, I couldn't find. Will try to implement this in the next PR if possible.

**Pull request closed without merging**